libusb/hid.c

```diff
@@ -548,11 +551,100 @@ int HID_API_EXPORT hid_exit(void)
 	return 0;
 }
 
+
+static void fill_device_info_for_device(libusb_device_handle *handle, struct hid_device_info * cur_dev, struct libusb_device_descriptor *desc)
+{
+	/* VID/PID */
+	cur_dev->vendor_id = desc->idVendor;
+	cur_dev->product_id = desc->idProduct;
+
+	/* Release Number */
+	cur_dev->release_number = desc->bcdDevice;
+
+	if (handle) {
+		/* Serial Number */
+		if (desc->iSerialNumber > 0)
+			cur_dev->serial_number =
+				get_usb_string(handle, desc->iSerialNumber);
+
+		/* Manufacturer and Product strings */
+		if (desc->iManufacturer > 0)
+			cur_dev->manufacturer_string =
+				get_usb_string(handle, desc->iManufacturer);
+		if (desc->iProduct > 0)
+			cur_dev->product_string =
+				get_usb_string(handle, desc->iProduct);
+
+#ifdef INVASIVE_GET_USAGE
+{
+		/*
+		This section is removed because it is too
+		invasive on the system. Getting a Usage Page
+		and Usage requires parsing the HID Report
+		descriptor. Getting a HID Report descriptor
+		involves claiming the interface. Claiming the
+		interface involves detaching the kernel driver.
+		Detaching the kernel driver is hard on the system
+		because it will unclaim interfaces (if another
+		app has them claimed) and the re-attachment of
+		the driver will sometimes change /dev entry names.
+		It is for these reasons that this section is
+		#if 0. For composite devices, use the interface
+		field in the hid_device_info struct to distinguish
+		between interfaces. */
+			unsigned char data[256];
+#ifdef DETACH_KERNEL_DRIVER
+			int detached = 0;
+			/* Usage Page and Usage */
+			res = libusb_kernel_driver_active(handle, interface_num);
+			if (res == 1) {
+				res = libusb_detach_kernel_driver(handle, interface_num);
+				if (res < 0)
+					LOG("Couldn't detach kernel driver, even though a kernel driver was attached.\n");
+				else
+					detached = 1;
+			}
+#endif
+			res = libusb_claim_interface(handle, interface_num);
+			if (res >= 0) {
+				/* Get the HID Report Descriptor. */
+				res = libusb_control_transfer(handle, LIBUSB_ENDPOINT_IN|LIBUSB_RECIPIENT_INTERFACE, LIBUSB_REQUEST_GET_DESCRIPTOR, (LIBUSB_DT_REPORT << 8)|interface_num, 0, data, sizeof(data), 5000);
+				if (res >= 0) {
+					unsigned short page=0, usage=0;
+					/* Parse the usage and usage page
+						out of the report descriptor. */
```

Comment:
this is specifically 3 spaces, and not a tab, to align the text (content of the comment)
```suggestion
					/* Parse the usage and usage page
					   out of the report descriptor. */
```